app/src/services/cpk.ts

```diff
@@ -256,6 +263,56 @@ class CPKService {
       throw err
     }
   }
+
+  sellOutcomes = async ({
+    amount,
+    outcomeIndex,
+    marketMaker,
+    conditionalTokens,
+  }: CPKSellOutcomesParams): Promise<TransactionReceipt> => {
+    try {
+      const signer = this.provider.getSigner()
+      const account = await signer.getAddress()
+
+      const outcomeTokensToSell = await marketMaker.calcSellAmount(amount, outcomeIndex)
+      const collateralAddress = await marketMaker.getCollateralToken()
+
+      const transactions = [
+        {
```

Comment:
We should only add this transaction if necessary to reduce the gas used.

Reply:
I added a check to add this transaction in case is not already approve for all